lib/Serialization/ScanningLoaders.cpp

```diff
@@ -337,22 +330,15 @@ llvm::ErrorOr<ModuleDependencyInfo> SwiftModuleScanner::scanBinaryModuleFile(
   return std::move(dependencies);
 }
 
-ModuleDependencyVector
+SwiftModuleScannerQueryResult
 SwiftModuleScanner::lookupSwiftModule(Identifier moduleName,
                                       bool isTestableImport) {
-  // When we exit, ensure we clear dependencies discovered on this query
-  SWIFT_DEFER { dependencies = std::nullopt; };
-
   ImportPath::Module::Builder builder(moduleName);
   auto modulePath = builder.get();
-  // Check whether there is a module with this name that we can import.
-  if (canImportModule(modulePath, SourceLoc(), nullptr, isTestableImport)) {
-    ModuleDependencyVector moduleDependnecies;
-    moduleDependnecies.push_back(std::make_pair(
-        ModuleDependencyID{moduleName.str().str(), dependencies->getKind()},
-        *(dependencies)));
-    return moduleDependnecies;
-  }
-
-  return {};
+  // Execute the check to determine whether there is a module with this name
+  // that we can import. This check will populate the result fields if one is
+  // found.
+  canImportModule(modulePath, SourceLoc(), nullptr, isTestableImport);
+  return SwiftModuleScannerQueryResult(
+      std::move(foundDependencyInfo), std::move(incompatibleCandidates));
```

Comment:
I think a `SWIFT_DEFER` to reset the value is still needed here. `std::move` left those global state in a valid but undefined state so the next lookup try to add to those results, they are undefined.

Reply:
Ah, got it, thank you.